### DISCUSSION TITLE
Multigraph Data Structure coding for GDAS NAS

### DISCUSSION THREAD

**Discussion by buttercutter** [pyg-team/pytorch_geometric#2636] [Q&A]
how to code the following graph structure ?
which kind of data structure is suitable in this case for both forward inference and backward propagation ?

![image](https://user-images.githubusercontent.com/3324659/119598844-4ae4d200-be16-11eb-8927-77fe3fc91ef2.png)

https://github.com/D-X-Y/AutoDL-Projects/issues/99#issuecomment-835713058

https://gist.github.com/promach/b6f526c56e20f029d68e6f9041c3f5c0#file-gdas-py-L106-L107

**Comment by rusty1s** [comment]
Representing multigraphs is well supported by simply duplicating edges in `edge_index`. Does that answer your question?

**Reply by buttercutter** [reply]
However, does the multigraph have support for both forward inference and backward propagation operations ?

**Reply by rusty1s** [reply]
`edge_index` is a `LongTensor`, so it cannot track any gradients. If you want to do that, I suggest to let gradients flow through an additional `edge_weight` tensor that determines the strengths of each edge.
